### PR TITLE
Make Normalize input transform input column specific

### PR DESCRIPTION
**Pull request by jduerholt**
## Motivation

This PR adds the possibility to apply a `Normalize` input transform only to specific input columns specified via a list of indices (in analogy to the `Log10` transform). This makes it possible to combine it with other transforms or intentionally ignore inputs in the transform.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/pytorch/botorch/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Unit tests.

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/pytorch/botorch, and link to your PR here.)

### CODE REVIEW

File: botorch/models/transforms/input.py [hunk 1]
```diff
@@ -340,6 +343,21 @@ def __init__(
                 zero errors.
         """
         super().__init__()
+        if (indices is None) or (len(indices) == 0):
+            self.register_buffer(
+                "indices", torch.tensor(list(range(d)), dtype=torch.long)
+            )
```
Comment (Balandat) [inline botorch/models/transforms/input.py:349]:
```suggestion
            self.register_buffer("indices", torch.arange(d))
```

Comment (Balandat) [inline botorch/models/transforms/input.py:349]: I wonder if a slightly better design would be to just not register an `indices` buffer and then then we just gate the indexing in the `_transform` and `_untransform` methods by sth like `if hasattr(self, "indices")):`. That would have the advantage of not doing a nullop clone + indexing operation in the all-indices case, which I anticipate to be the most common one.

Comment (jduerholt) [reply botorch/models/transforms/input.py]: Good, idea. I will change it accordingly.

File: botorch/models/transforms/input.py [hunk 2]
```diff
@@ -340,6 +343,21 @@ def __init__(
                 zero errors.
         """
         super().__init__()
+        if (indices is None) or (len(indices) == 0):
```
[the remark this answers is not in the document]
Comment (jduerholt) [reply botorch/models/transforms/input.py:346]: I will error it out.

**Comment by jduerholt** [comment]
Hi @Balandat, 

I implemented your suggested changes. `indices` are now not anymore registered as a buffer. Instead it is gated by a `hasattr` call. This change make the `equals` method a bit more complicated, but I think this is ok. 

Just have a look yourself ;-)

Best,

Johannes

P.S.: I wish you a happy new year.

**Comment by Balandat** [comment]
Happy new year to you as well, and thanks for updating the PR! Will merge this soon.
